test/e2e_env/universal/meshservice/meshservice.go

```diff
@@ -0,0 +1,84 @@
+package meshservice
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/universal"
+)
+
+func MeshService() {
+	meshName := "mesh-service"
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(MeshUniversal(meshName)).
+			Install(TestServerUniversal(
+				"first-test-server",
+				meshName,
+				WithArgs([]string{"echo"}),
+				WithServiceName("first-test-server"),
+				WithAdditionalTags(map[string]string{
+					"app": "test-server",
+				}))).
+			Install(DemoClientUniversal(AppModeDemoClient, meshName, WithTransparentProxy(true))).
+			Setup(universal.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEachFailure(func() {
+		DebugUniversal(universal.Cluster, meshName)
+	})
+
+	E2EAfterAll(func() {
+		Expect(universal.Cluster.DeleteMeshApps(meshName)).To(Succeed())
+		Expect(universal.Cluster.DeleteMesh(meshName)).To(Succeed())
+	})
+
+	It("should be able to create and use MeshService with HostnameGenerator", func() {
+		generator := `
+type: HostnameGenerator
+mesh: mesh-service
+name: basic
+labels:
+  kuma.io/origin: zone
+spec:
+  template: "{{ .Name }}.mesh"
```

Comment:
can we do anything other than `.mesh` to avoid "clash" with existing things?

Reply:
added `.meshservice.mesh`